Microsoft.Toolkit.Uwp.UI.Controls/Carousel/Carousel.cs

```diff
@@ -294,7 +292,7 @@ public Carousel()
                 var itemsPanel = carouselControl.GetItemsPanel();
 
                 // Prioritize the SelectedItem over the SelectedIndex
-                if (SelectedItem != null && Items[carouselControl.SelectedIndex] != SelectedItem)
+                if (SelectedItem != null && (Items.Count() <= carouselControl.SelectedIndex || Items[carouselControl.SelectedIndex] != SelectedItem))
```

Comment:
I think this is to resolve the issue if you have a collection of 10 elements, and have the 9th selected.  Your selected index is 8.  If you then swap out the collection to one that has 5 elements, then this line would throw an exception as it was trying to index 8 into a list now containing 5 items.  

The added check now forces it down into this code in that case to see if the originally selected item is in the new list or not.